src/blog/2025/06/flowfuse-forms-easy-data-collection-factory-floor.md

```diff
@@ -0,0 +1,336 @@
+- **Basic Node-RED Knowledge:** You are familiar with creating and deploying basic flows in Node-RED. If not, consider taking the [Node-RED Fundamentals Course](https://node-red-academy.learnworlds.com/course/node-red-getting-started) *sponsored by FlowFuse.*
+
+## Building Forms in FlowFuse Dashboard
+
+The FlowFuse Dashboard makes it easy to build interactive industrial applications using drag-and-drop components—no coding required. One of these components is the [ui_form](https://dashboard.flowfuse.com/nodes/widgets/ui-form.html) widget, which allows you to create versatile forms within your application. The ui_form widget supports a wide range of input types, including text fields, number inputs, date pickers, multi-line text areas, dropdowns, and checkboxes. A key benefit of this widget is that you can configure the form fields either statically (with predefined values) or dynamically (updated through your Node-RED flow), depending on your application’s needs.
+
+To get started with building a form, follow these steps to add and configure the **ui_form** widget in your FlowFuse Dashboard.
+
+### Adding and Configuring the Form Widget
+
+1. Drag the **ui_form** widget onto the canvas.
+2. Double-click on the widget and create a new group for it with the correct page configuration to render it.
+3. Set the correct size (width and height) according to your preferences.
+4. Enter the label for the form.
+
+Now that we have completed all the basic and necessary configurations for the form, let’s add the input elements.
+
+### Adding Input Fields to the Form Statically
+
+The `ui_form` widget supports various input element types that can be tailored to specific use cases — from collecting simple text to selecting dates or choosing from predefined options.
+
+**To add input elements:**
+
+1. Click the **+ add** button in the widget’s configuration dialog.
+2. A new configuration row will appear for the element.
+3. Configure each input element with the following fields:
+
+- **Label**: This is the visible label for the field shown to the user.
+- **Name**: A unique key used in the message payload (e.g., `msg.payload.firstname`) when the form is submitted.
+- **Type**: Select the input type. Supported types include:
+  - **Text**: For short text inputs (e.g., name, city).
+  - **Number**: For numeric inputs (e.g., age, price).
+  - **Date**: For selecting a date.
+  - **Text Area**: For longer free-form text.
+  - **Dropdown**: For selecting from a list of predefined values.  
+    _We will cover how to add options to the dropdown field in a later section._
+  - **Checkbox**: For boolean values (checked or unchecked).
+- **Required**: Check this box to make the field mandatory. The form cannot be submitted unless this field is filled.
+- **Row**:: If Multiline is selected, this defines the number of visible rows in the text area.
+
+![Adding Form Elements](./images/adding-form-element.gif){data-zoomable}
```

Comment:
move this to the first thing below the "Adding Input Fields to the Form Statically" headline. It shows how easy it is, and the text can be intimidating.